osquery/tables/system/posix/augeas.cpp

```diff
@@ -130,50 +81,71 @@ void matchAugeasPattern(augeas* aug,
     std::string node(matches[i]);
     free(matches[i]);
 
-    Row r;
-    const char* value = nullptr;
-    int result = aug_get(aug, node.c_str(), &value);
-    if (result == 1) {
-      r["node"] = node;
-
-      if (value != nullptr) {
-        r["value"] = value;
-      }
-
-      if (!use_path) {
-        r["path"] = getSpanInfo(aug, node, context);
-      } else {
-        r["path"] = pattern;
-      }
-
-      r["label"] = getLabelInfo(aug, node, context);
-
-      results.push_back(r);
-    } else if (result < 1) {
+    const char *value = nullptr, *label = nullptr;
+    char* file = nullptr;
+    result = aug_ns_attr(aug, "matches", i, &value, &label, &file);
+    if (result == -1) {
       reportAugeasError(aug);
+      return;
+    }
+
+    std::string path;
+    if (file != nullptr) {
+      path = file;
+      path = path.substr(6);
```

Comment:
If `path.size() < 6` this will throw an exception.

Reply:
I'm pretty sure it's not possible. All values begin with `/files` which is 6 characters long.